govcd/nsxt_alb_service_engine_group_assignment.go

```diff
@@ -108,6 +100,31 @@ func (vcdClient *VCDClient) GetAlbServiceEngineGroupAssignmentByName(name string
 	return foundGroup, nil
 }
 
+// GetFilteredAlbServiceEngineGroupAssignmentByName will get all ALB Service Engine Group assignments based on filters
+// provided in queryParameters additionally will filter by name locally because VCD does not support server side
+// filtering by name.
+func (vcdClient *VCDClient) GetFilteredAlbServiceEngineGroupAssignmentByName(name string, queryParameters url.Values) (*NsxtAlbServiceEngineGroupAssignment, error) {
+	// Filtering by Service Engine Group name is not supported on API therefore filtering is done locally
+	allServiceEngineGroupAssignments, err := vcdClient.GetAllAlbServiceEngineGroupAssignments(queryParameters)
+	if err != nil {
+		return nil, err
+	}
+
+	var foundGroup *NsxtAlbServiceEngineGroupAssignment
+
+	for _, serviceEngineGroupAssignment := range allServiceEngineGroupAssignments {
+		if serviceEngineGroupAssignment.NsxtAlbServiceEngineGroupAssignment.ServiceEngineGroupRef.Name == name {
+			foundGroup = serviceEngineGroupAssignment
```

Comment:
Is the name unique? in case we don't return only first one found

Reply:
It must be unique:
<img width="660" alt="image" src="https://user-images.githubusercontent.com/15804230/146799999-2f5fc63e-e3f1-493c-a520-b34ca4078d05.png">